core/src/main/scala/caliban/schema/Schema.scala

```diff
@@ -356,7 +359,11 @@ trait DerivationSchema[R] {
 
     override def resolve(value: T): Step[R] =
       if (ctx.isObject) PureStep(EnumValue(getName(ctx)))
-      else ObjectStep(getName(ctx), ctx.parameters.map(p => p.label -> p.typeclass.resolve(p.dereference(value))).toMap)
+      else if (ctx.isValueClass && ctx.parameters.nonEmpty) {
+        val head = ctx.parameters.head
+        head.typeclass.resolve(head.dereference(value))
```

Comment:
For some reason, the above compiles but this doesn't: `ctx.parameters.head.typeclass.resolve(ctx.parameters.head.dereference(value))`

We get:
```
[error] /Users/palan/code/caliban/core/src/main/scala/caliban/schema/Schema.scala:363:78: type mismatch;
[error]  found   : magnolia.ReadOnlyParam[DerivationSchema.this.Typeclass,T]#PType
[error]  required: _1.PType where val _1: magnolia.ReadOnlyParam[DerivationSchema.this.Typeclass,T]
[error]         ctx.parameters.head.typeclass.resolve(ctx.parameters.head.dereference(value))
[error]                                                                              ^
```